.github/workflows/clang-tidy.yml

```diff
@@ -0,0 +1,88 @@
+name: CLANG_TIDY
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+
+#addons:
+#  homebrew:
+#    packages:
+#    - git
+#  apt:
+#    sources:
+#    - llvm-toolchain-xenial-8
+#    - ubuntu-toolchain-r-test
+#    packages:
+#    - llvm-8
+#    - g++-8
+#    - clang-8
+#    - git
+#    - python3
+#    - python3-pip
+#    - python3-setuptools
+#    - clang-tidy-8
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [linux]
+        compiler: [{
+          "cc": "gcc",
+          "cxx": "g++"
+        }]
+    steps:
+    - uses: actions/checkout@v1
+      name: checkout
+      with:
+        submodules: true
+        clean: true
+        fetch-depth: 1
+    - name: install
+      run: |
+        # install python pip3 deps
+        if [ "$TRAVIS_OS_NAME" = "linux" ]; then
+          LLVM_BIN=/usr/lib/llvm-8/bin
+          if [ -d $LLVM_BIN ]; then
+            export PATH="$PATH:$LLVM_BIN"
+            echo "$LLVM_BIN added to PATH"
+          else
+            echo "$LLVM_BIN is absent"
+          fi
+        elif [ "$TRAVIS_OS_NAME" = "osx" ]; then
+          pip3 install --user pyyaml
+        fi
+        sudo python3 -m pip install --upgrade pip
+        sudo python3 -m pip install scikit-build
+        sudo python3 -m pip install cmake requests gitpython gcovr
+        export PATH="$HOME/.cargo/bin:$PATH"
+        curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain nightly-2019-10-16 --profile minimal
+    - name: before_script
+      run: cmake --version
+        python -V
+        python3 -V
+        pip -V
+        pip3 -V
+        gcovr --version || true
+        echo $PATH
+    - name: script
+      run: export CC=${gcc}
+        export CXX=${g++}
+        export CC=${_CC}
+        export CXX=${_CXX}
+        export PATH="$HOME/.cargo/bin:$PATH"
+        export LLVM_ROOT=/usr/lib/llvm-8
+        export PATH=${LLVM_ROOT}/bin:${LLVM_ROOT}/share/clang:${PATH}
+        sudo update-alternatives --install /usr/bin/clang-tidy clang-tidy /usr/bin/clang-tidy-8 90
+        cmake . -Bbuild
+        cmake --build build --target generated
+        housekeeping/clang-tidy.sh build
+#        cmake . -Bbuild -DCOVERAGE=${COVERAGE:-OFF} -DCMAKE_TOOLCHAIN_FILE=${TOOLCHAIN:-cmake/toolchain/cxx17.cmake}
+#        cmake --build build -- -j2
+#        cmake --build build --target test
```

Comment:
remove comments.
also, do not build project during this step. you only need to execute `cmake` to generate `compile_commands.json`